apps/deploy-web/src/components/remote-deploy/helper-functions.ts

```diff
@@ -0,0 +1,137 @@
+import { Control, UseFormSetValue } from "react-hook-form";
+import { nanoid } from "nanoid";
+
+import { SdlBuilderFormValuesType, ServiceType } from "@src/types";
+import { Owner } from "@src/types/remotedeploy";
+export const ciCdTemplateId = "akash-network-awesome-akash-automatic-deployment-CICD-template";
+export type ServiceControl = Control<SdlBuilderFormValuesType>;
+export type ServiceSetValue = UseFormSetValue<SdlBuilderFormValuesType>;
+export type OAuth = "github" | "gitlab" | "bitbucket";
+export const PROXY_API_URL_AUTH = "https://proxy-console-github.vercel.app";
+
+export const protectedEnvironmentVariables = {
+  REPO_URL: "REPO_URL",
+  BRANCH_NAME: "BRANCH_NAME",
+  ACCESS_TOKEN: "ACCESS_TOKEN",
+  BUILD_DIRECTORY: "BUILD_DIRECTORY",
+  BUILD_COMMAND: "BUILD_COMMAND",
+  NODE_VERSION: "NODE_VERSION",
+  CUSTOM_SRC: "CUSTOM_SRC",
+  COMMIT_HASH: "COMMIT_HASH",
+  GITLAB_PROJECT_ID: "GITLAB_PROJECT_ID",
+  GITLAB_ACCESS_TOKEN: "GITLAB_ACCESS_TOKEN",
+  BITBUCKET_ACCESS_TOKEN: "BITBUCKET_ACCESS_TOKEN",
+  BITBUCKET_USER: "BITBUCKET_USER",
+  DISABLE_PULL: "DISABLE_PULL",
+  GITHUB_ACCESS_TOKEN: "GITHUB_ACCESS_TOKEN",
+  FRONTEND_FOLDER: "FRONTEND_FOLDER",
+  INSTALL_COMMAND: "INSTALL_COMMAND"
+};
+
+export const REDIRECT_URL = `${process.env.NEXT_PUBLIC_REDIRECT_URI}?step=edit-deployment&type=github`;
+export function appendEnv(key: string, value: string, isSecret: boolean, setValue: ServiceSetValue, services: ServiceType[]) {
+  const previousEnv = services[0]?.env || [];
+  if (previousEnv.find(e => e.key === key)) {
+    previousEnv.map(e => {
+      if (e.key === key) {
+        e.value = value;
+        e.isSecret = isSecret;
+
+        return e;
+      }
+      return e;
+    });
+  } else {
+    previousEnv.push({ id: nanoid(), key, value, isSecret });
+  }
+  setValue("services.0.env", previousEnv);
+}
+
+export function removeEnv(key: string, setValue: ServiceSetValue, services: ServiceType[]) {
+  const previousEnv = services[0]?.env || [];
+  const newEnv = previousEnv.filter(e => e.key !== key);
+  setValue("services.0.env", newEnv);
+}
+
+export const removeInitialUrl = (url?: string) => {
+  return url?.split("/").slice(-2).join("/");
+};
+
+export interface RepoType {
+  name: string;
+  id?: string;
+  default_branch: string;
+  html_url: string;
+  userName?: string;
+  private: boolean;
+  owner?: Owner;
+}
+
+export const isRedeployImage = (yml: string, ciCdYml?: string) => {
```

Comment:
CICD is usually used as an acronym of the same case.

```suggestion
export const isRedeployImage = (yml: string, cicdYml?: string) => {
```